src/android/InAppBrowser.java

```diff
@@ -966,21 +1033,56 @@ public boolean onKey(View v, int keyCode, KeyEvent event) {
                     }
                 });
 
+                // Navigation bar layout
+                GradientDrawable navigationBarBorderBackground = new GradientDrawable();
+                navigationBarBorderBackground.setColor(toolbarColor);
+                navigationBarBorderBackground.setStroke(1, android.graphics.Color.parseColor("#E9E6E1"));
+                RelativeLayout navigationBar = new RelativeLayout(cordova.getActivity());
+                if(android.os.Build.VERSION.SDK_INT < android.os.Build.VERSION_CODES.JELLY_BEAN) {
+                    navigationBar.setBackgroundDrawable(navigationBarBorderBackground);
+                } else {
+                    navigationBar.setBackground(navigationBarBorderBackground);
+                }
+                navigationBar.setLayoutParams(new RelativeLayout.LayoutParams(LayoutParams.MATCH_PARENT, this.dpToPixels(44)));
+                navigationBar.setPadding(this.dpToPixels(2), this.dpToPixels(2), this.dpToPixels(2), this.dpToPixels(2));
+                navigationBar.setVerticalGravity(Gravity.CENTER);
+                navigationBar.setHorizontalGravity(Gravity.LEFT);
+
+                // Add page title for navigation bar
+                pageTitleTextView = new TextView(cordova.getActivity());
+                LinearLayout.LayoutParams pageTitleTextViewLayoutParams = new LinearLayout.LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.MATCH_PARENT);
+                pageTitleTextView.setLayoutParams(pageTitleTextViewLayoutParams);
+                pageTitleTextView.setText("Loading...");
+                pageTitleTextView.setTextSize(bannerTextSize);
+                pageTitleTextView.setTextColor(android.graphics.Color.parseColor(navigationButtonColor));
+                pageTitleTextView.setTypeface(null, Typeface.BOLD);
+                int lockResId = activityRes.getIdentifier("lock", "drawable", cordova.getActivity().getPackageName());
+                Drawable lockIcon = activityRes.getDrawable(lockResId);
+                pageTitleTextView.setCompoundDrawablesWithIntrinsicBounds(lockIcon, null, null, null);
+                pageTitleTextView.setCompoundDrawablePadding(this.dpToPixels(8));
+                pageTitleTextView.setGravity(Gravity.CENTER_VERTICAL);
+                pageTitleTextView.setPadding(this.dpToPixels(20), 0, 0, 0);
 
                 // Header Close/Done button
                 int closeButtonId = leftToRight ? 1 : 5;
                 View close = createCloseButton(closeButtonId);
-                toolbar.addView(close);
```

Comment:
Should the above lines be deleted too or should this get added only when `showNavigationBar` is not 'yes'?

Reply:
This button is initialized here and gets added to the view later in line `1215` or `1224`. I left it here because we want the same button regardless of where it ends up.